x-pack/plugins/ingest_pipelines/public/application/components/pipeline_processors_editor/components/manage_processor_form/processors/join.tsx

```diff
@@ -0,0 +1,71 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+import React, { FunctionComponent } from 'react';
+import { i18n } from '@kbn/i18n';
+import { FormattedMessage } from '@kbn/i18n/react';
+import { EuiCode } from '@elastic/eui';
+
+import { FIELD_TYPES, fieldValidators, UseField, Field } from '../../../../../../shared_imports';
+
+import { FieldsConfig } from './shared';
+import { FieldNameField } from './common_fields/field_name_field';
+import { TargetField } from './common_fields/target_field';
+
+const { emptyField } = fieldValidators;
+
+const fieldsConfig: FieldsConfig = {
+  /* Required fields config */
+  separator: {
+    type: FIELD_TYPES.TEXT,
+    label: i18n.translate('xpack.ingestPipelines.pipelineEditor.joinForm.separatorFieldLabel', {
+      defaultMessage: 'Separator',
+    }),
+    deserializer: String,
+    helpText: i18n.translate(
+      'xpack.ingestPipelines.pipelineEditor.joinForm.separatorFieldHelpText',
+      {
+        defaultMessage: 'Separator character.',
+      }
+    ),
+    validations: [
+      {
+        validator: emptyField(
+          i18n.translate('xpack.ingestPipelines.pipelineEditor.joinForm.separatorRequiredError', {
+            defaultMessage: 'A separator value is required.',
+          })
+        ),
+      },
+    ],
+  },
+};
+
+export const Join: FunctionComponent = () => {
+  return (
+    <>
+      <FieldNameField
+        helpText={i18n.translate(
+          'xpack.ingestPipelines.pipelineEditor.joinForm.fieldNameHelpText',
+          { defaultMessage: 'Field containing array values to join.' }
```

Comment:
Period isn't needed.
```suggestion
          { defaultMessage: 'Field containing array values to join' }
```